src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesGraphProjectContextProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.VS.Extensibility;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    /// <summary>
+    /// Global scope contract that provides information about project level 
+    /// dependencies graph contexts.
+    /// </summary>
+    [Export(typeof(IDependenciesGraphProjectContextProvider))]
+    [AppliesTo(ProjectCapability.AlwaysAvailable)]
+    public class DependenciesGraphProjectContextProvider : IDependenciesGraphProjectContextProvider
```

Comment:
I think we should think about when this is turned on  - this is going to immediately regress RPS because we're loaded in all projects. We should put this behind a capability.